### PR TITLE
Support ghc 9.12

### PR DESCRIPTION
**Pull request by erikd**

### CODE REVIEW

**Comment by andreasabel** [comment]
@erikd Could you also add 9.12.0 to `tested-with` and update the Haskell CI.
After installing my fork https://github.com/andreasabel/haskell-ci run `haskell-ci regenerate`.
You might have to bump the other `tested-with` entries to the latest GHC minor versions  Also GHC 7 isn't supported any more by Haskell Ci.

**Comment by erikd** [comment]
@andreasabel Done!

**Comment by erikd** [comment]
@andreasabel Done

File: .github/workflows/haskell-ci.yml [hunk 1]
```diff
@@ -6,11 +6,11 @@
...
 #   haskell-ci regenerate
 #
-# For more information, see https://github.com/andreasabel/haskell-ci
+# For more information, see https://github.com/haskell-CI/haskell-ci
 #
-# version: 0.19.20240403
+# version: 0.19.20240708
 #
-# REGENDATA ("0.19.20240403",["github","parallel.cabal"])
+# REGENDATA ("0.19.20240708",["github","parallel.cabal"])
```
Comment (andreasabel) [inline .github/workflows/haskell-ci.yml:13]: This version of haskell-ci is too old, does not know about GHC 9.12 yet.

File: parallel.cabal [hunk 2]
```diff
@@ -12,9 +12,11 @@ category:       Control, Parallelism
 build-type:     Simple
 
 tested-with:
-  GHC == 9.10.0
+  -- Not supported yet
+  -- GHC == 9.12.0
```
Comment (andreasabel) [inline parallel.cabal:16]: My fork supports 9.12.0, I think on the original Haskell CI the same is called 9.12.1.

**Review by andreasabel — CHANGES_REQUESTED** [review]
Thanks!

I think it would be good if CI tested the new base bound, so GHC 9.12 should be included in CI.
Latest builds of Haskell CI do support the alpha1 version.

File: parallel.cabal (hunk 2 above)
Comment (erikd) [reply parallel.cabal:16]: I am currently using the one from https://github.com/haskell-CI/haskell-ci @ master :confused:

Comment (erikd) [reply parallel.cabal]: Ok, now used your branch.

**Review by andreasabel — APPROVED** [review]
Great!

**Comment by andreasabel** [comment]
Published as revision 9: https://hackage.haskell.org/package/parallel-3.2.2.0/revisions/

**Pull request merged**